packages/hint-compat-api/src/helpers/compat-api.ts

```diff
@@ -0,0 +1,152 @@
+// Waiting for this PR https://github.com/mdn/browser-compat-data/pull/3004
+const bcd: CompatData = require('mdn-browser-compat-data');
+
+import { forEach } from 'lodash';
+import { BrowserSupportCollection, MDNTreeFilteredByBrowsers } from '../types';
+import { CompatData, CompatStatement, SupportStatement, SimpleSupportStatement } from '../types-mdn.temp'; // Temporal
+
+type CompatNamespace = 'css' | 'javascript' | 'html';
+
+export class CompatApi {
+    public compatDataApi: MDNTreeFilteredByBrowsers; // Any because no types by the moment, check line 1
+    private browsers: BrowserSupportCollection;
+
+    public constructor(namespaceName: CompatNamespace, browsers: BrowserSupportCollection, isCheckingNotBroadlySupported = false) {
+        this.browsers = browsers;
+        this.compatDataApi = bcd[namespaceName];
+        this.compatDataApi = this.applyBrowsersConfiguration(isCheckingNotBroadlySupported);
+    }
+
+    private applyBrowsersConfiguration(isCheckingNotBroadlySupported = false): any {
+        const compatDataApi = {} as MDNTreeFilteredByBrowsers;
+
+        forEach(this.compatDataApi, (namespaceFeaturesValues, namespaceFeaturesKey) => {
+            const namespaceFeatures = {} as CompatStatement & MDNTreeFilteredByBrowsers;
+
+            forEach(namespaceFeaturesValues, (featureValue, featureKey) => {
+                const typedFeatureValue = featureValue as CompatStatement & MDNTreeFilteredByBrowsers;
+
+                if (!this.isFeatureRequiredToTest(typedFeatureValue, isCheckingNotBroadlySupported)) {
+                    return;
+                }
+
+                namespaceFeatures[featureKey] = typedFeatureValue;
+            });
+
+            compatDataApi[namespaceFeaturesKey] = namespaceFeatures;
+        });
+
+        return compatDataApi;
+    }
+
+    public getPrefix(name: string): [string, string] | [undefined, string] {
```

Comment:
Slightly more concise:
```suggestion
    public getPrefix(name: string): [string | undefined, string] {
```